pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -136,3 +141,74 @@ func handleAttestCertDelete(ctxArg interface{}, key string,
 	log.Infof("handleAttestCertDelete for %s", status.Key())
 	return
 }
+
+// Run a task certificate post task, on change trigger
+func certsTask(ctx *zedagentContext, triggerCerts <-chan struct{}) {
+	iteration := 0
+	log.Infoln("starting certs publish task")
+	publishCerts(ctx, iteration)
+
+	// Run a periodic timer so we always update StillRunning
+	stillRunning := time.NewTicker(25 * time.Second)
+	agentlog.StillRunning(agentName+"certs", warningTime, errorTime)
+
+	for {
+		select {
+		case <-triggerCerts:
+			start := time.Now()
+			iteration++
+			publishCerts(ctx, iteration)
+			pubsub.CheckMaxTimeTopic(agentName+"certs", "publishCerts", start,
+				warningTime, errorTime)
+
+		case <-stillRunning.C:
+		}
+		agentlog.StillRunning(agentName+"certs", warningTime, errorTime)
+	}
+}
+
+// prepare the certs list proto message
+func publishCerts(ctx *zedagentContext, iteration int) {
+	var attestReq = &attest.ZAttestReq{}
+
+	startPubTime := time.Now()
```

Comment:
Nit, but the caller already mesures and reports long times using CheckMaxTimeTopic, so why do you need it here as well?

Reply:
Sure done.